src/core/src/pass/serialize.cpp

```diff
@@ -1262,77 +1262,70 @@ bool pass::Serialize::run_on_model(const std::shared_ptr<ov::Model>& model) {
-        const auto& binPath_ref = ov::util::string_to_wstring(m_binPath);
-        std::string message_bin = "Can't open bin file.";
-        std::string message_xml = "Can't open xml file.";
+            
+ #if defined(OPENVINO_ENABLE_UNICODE_PATH_SUPPORT) && defined(_WIN32)
+    const auto& xmlPath_ref = ov::util::string_to_wstring(m_xmlPath);
+    const auto& binPath_ref = ov::util::string_to_wstring(m_binPath);
+    std::string message_bin = "Can't open bin file.";
+    std::string message_xml = "Can't open xml file.";
 #else
-        const auto& xmlPath_ref = m_xmlPath;
-        const auto& binPath_ref = m_binPath;
-        std::string message_bin = "Can't open bin file: \"" + binPath_ref + "\"";
-        std::string message_xml = "Can't open xml file: \"" + xmlPath_ref + "\"";
+    const auto& xmlPath_ref = m_xmlPath;
+    const auto& binPath_ref = m_binPath;
+    std::string message_bin = "Can't open bin file: \"" + binPath_ref + "\"";
+    std::string message_xml = "Can't open xml file: \"" + xmlPath_ref + "\"";
 #endif
-        auto xmlDir = ov::util::get_directory(xmlPath_ref);
-        if (xmlDir != xmlPath_ref)
-            ov::util::create_directory_recursive(xmlDir);
-
-        std::ofstream bin_file(binPath_ref, std::ios::out | std::ios::binary);
-        OPENVINO_ASSERT(bin_file, message_bin);
-
-        // create xml file
-        std::ofstream xml_file(xmlPath_ref, std::ios::out);
-        OPENVINO_ASSERT(xml_file, message_xml);
-
-        try {
-            serializeFunc(xml_file, bin_file, model, m_version);
-        } catch (const ov::AssertFailure&) {
-            // optimization decision was made to create .bin file upfront and
-            // write to it directly instead of buffering its content in memory,
-            // hence we need to delete it here in case of failure
-            xml_file.close();
-            bin_file.close();
-            std::ignore = std::remove(m_xmlPath.c_str());
-            std::ignore = std::remove(m_binPath.c_str());
-            throw;
+
+    // Ensure the xml directory exists.
+    auto xmlDir = ov::util::get_directory(xmlPath_ref);
+    if (xmlDir != xmlPath_ref)
+        ov::util::create_directory_recursive(xmlDir);
+
+    // ---- Disk Space Check Start ----
+    {
+        #include <filesystem>
+        namespace fs = std::filesystem;
+        constexpr std::uintmax_t MIN_DISK_SPACE_REQUIRED = 50 * 1024 * 1024; // adjust as needed
+
+        // Check available space for bin file directory
+        fs::path binDir(binPath_ref);
+        if (binDir.has_parent_path())
+            binDir = binDir.parent_path();
+        fs::space_info binSpace = fs::space(binDir);
+        if (binSpace.available < MIN_DISK_SPACE_REQUIRED) {
+            OPENVINO_ASSERT(false, "Insufficient disk space for bin file: \"" + binPath_ref +
+                                         "\". Available: " + std::to_string(binSpace.available) + " bytes.");
```

Comment:
```suggestion
            OPENVINO_ASSERT(binSpace.available > MIN_DISK_SPACE_REQUIRED, "Insufficient disk space for bin file: \"", binPath_ref,
                                         "\". Available: ", binSpace.available, " bytes.");
```
Put condition instead `false`